vendor/bat-native-ledger/src/bat/ledger/internal/state/client_state.cc

```diff
@@ -0,0 +1,558 @@
+
+  writer->String(kBootTimestampKey);
+  writer->Uint64(properties.boot_timestamp);
+
+  writer->String(kReconcileTimestampKey);
+  writer->Uint64(properties.reconcile_timestamp);
+
+  writer->String(kPersonaIdKey);
+  writer->String(properties.persona_id.c_str());
+
+  writer->String(kUserIdKey);
+  writer->String(properties.user_id.c_str());
+
+  writer->String(kRegistrarVkKey);
+  writer->String(properties.registrar_vk.c_str());
+
+  writer->String(kMasterUserTokenKey);
+  writer->String(properties.master_user_token.c_str());
+
+  writer->String(kPreFlightKey);
+  writer->String(properties.pre_flight.c_str());
+
+  writer->String(kFeeCurrencyKey);
+  writer->String(properties.fee_currency.c_str());
+
+  writer->String(kSettingsKey);
+  writer->String(properties.settings.c_str());
+
+  writer->String(kFeeAmountKey);
+  writer->Double(properties.fee_amount);
+
+  writer->String(kUserChangedFeeKey);
+  writer->Bool(properties.user_changed_fee);
+
+  writer->String(kDaysKey);
+  writer->Uint(properties.days);
+
+  writer->String(kRewardsEnabledKey);
+  writer->Bool(properties.rewards_enabled);
+
+  writer->String(kAutoContributeKey);
+  writer->Bool(properties.auto_contribute);
+
+  writer->String(kTransactionsKey);
+  writer->StartArray();
+  const TransactionState transaction_state;
+  for (const auto& transaction : properties.transactions) {
+    transaction_state.ToJson(writer, transaction);
+  }
+  writer->EndArray();
+
+  writer->String(kBallotsKey);
+  writer->StartArray();
+  const BallotState ballot_state;
+  for (const auto& ballot : properties.ballots) {
+    if (!ballot_state.ToJson(writer, ballot)) {
+      NOTREACHED();
+      continue;
+    }
+  }
+  writer->EndArray();
+
+  writer->String(kRuleSetKey);
+  writer->String(properties.rule_set.c_str());
+
+  writer->String(kRuleSetV2Key);
+  writer->String(properties.rule_set_v2.c_str());
+
+  writer->String(kBatchVotesKey);
+  writer->StartArray();
+  const BatchVotesState batch_votes_state;
+  for (const auto& batch_vote : properties.batch_votes) {
+    if (!batch_votes_state.ToJson(writer, batch_vote)) {
+      NOTREACHED();
+      continue;
+    }
+  }
+  writer->EndArray();
+
+  writer->String(kCurrentReconcilesKey);
+  writer->StartObject();
+  const CurrentReconcileState current_reconcile_state;
+  for (const auto& current_reconcile : properties.current_reconciles) {
+    writer->Key(current_reconcile.first.c_str());
+    if (!current_reconcile_state.ToJson(writer, current_reconcile.second)) {
+      NOTREACHED();
+      continue;
+    }
+  }
+  writer->EndObject();
+
+  writer->String(kWalletKey);
+  const WalletState wallet_state;
+  wallet_state.ToJson(writer, properties.wallet);
+
+  writer->String(kInlineTipsKey);
+  writer->StartObject();
+  for (const auto &inline_tip : properties.inline_tips) {
+    writer->String(inline_tip.first.c_str());
+    writer->Bool(inline_tip.second);
+  }
+  writer->EndObject();
+
+  writer->EndObject();
+
+  return true;
+}
```

Comment:
Would recommend a ticket to refactor/split this super-huge method into smaller methods.

Reply:
See below comment